drivers/mmcsd/mmcsd_sdio.c

```diff
@@ -451,6 +451,22 @@ static int mmcsd_get_scr(FAR struct mmcsd_state_s *priv, uint32_t scr[2])
 {
   int ret;
 
+#if defined(CONFIG_SDIO_DMA) && defined(CONFIG_ARCH_HAVE_SDIO_PREFLIGHT)
+  /* If we think we are going to perform a DMA transfer, make sure that we
+   * will be able to before we commit the card to the operation.
+   */
+
+  if ((priv->caps & SDIO_CAPS_DMASUPPORTED) != 0)
```

Comment:
This will only fail to read an SCR and not mount the card.

I think this logic belongs in the `if` on line 485 ish to drop to `SDIO_RECVSETUP(priv->dev, (FAR uint8_t *)scr, 8);` if it fails preflight.

Reply:
@davids5 
Modified as suggested.

By the way, what about other SDIO_DMARECVSETUPs in mmcsd_readsingle, mmcsd_readmultiple, mmcsd_read_csd, and mmcsd_general_cmd_read?